src/Search/SearchHandler.php

```diff
@@ -84,20 +84,31 @@ public function search(AdminInterface $admin, $term, $page = 0, $offset = 20)
         $datagridValues = $datagrid->getValues();
 
         $found = false;
+        $previousFilter = null;
         foreach ($datagrid->getFilters() as $filter) {
             /** @var FilterInterface $filter */
             $formName = $filter->getFormName();
 
             if ($filter->getOption('global_search', false)) {
                 $filter->setOption('case_sensitive', $this->caseSensitive);
-                $filter->setOption('or_group', $admin->getCode());
+
+                if ($filter instanceof ChainableFilterInterface) {
+                    if (null !== $previousFilter) {
+                        $filter->setPreviousFilter($previousFilter);
+                    }
+                } else {
+                    // NEXT_MAJOR: Remove this `else` block.
+                    $filter->setOption('or_group', $admin->getCode());
+                }
                 $filter->setCondition(FilterInterface::CONDITION_OR);
                 $datagrid->setValue($formName, null, $term);
                 $found = true;
             } elseif (isset($datagridValues[$formName])) {
                 // Remove any previously set filter that is not configured for the global search.
                 $datagrid->removeFilter($formName);
             }
+
+            $previousFilter = $filter;
```

Comment:
The previous filter should be the previous used for the global search, so this should be in the if part too.